edgeless_node/src/resources/dda/mod.rs

```diff
@@ -80,7 +80,8 @@ impl DDAResource {
         let dda_sub_array = match dcs {
             Ok(dda_array) => dda_array,
             Err(err) => {
-                eprintln!("Error parsing input dda_com_subscription_mapping JSON: {}", err);
+                log::error!("Error parsing input dda_com_subscription_mapping JSON: {}", err);
+                //eprintln!("Error parsing input dda_com_subscription_mapping JSON: {}", err);
                 process::exit(1);
```

Comment:
Uhm... I know that there has already been a discussion thread on this and we agreed to postpone the discussion to the code camp in Chania and all, but this `process::exit()` makes me feel _very uncomfortable_ ☹️ 

The error could result from parsing input that is user-provided, so this does not look like an assertion/invariant to me (in case `panic!()` would be more appropriate, anyway).

I suggest to remove all `process::exit()` instructions. If the input is invalid, an error will be logged and that's all that will happen for now (i.e., the resource started will be inconsistent). This is far from ideal, I agree, but at least a single misconfiguration on a workflow will not jeopardize the entire node.

Does this make sense?

Reply:
I will comment out exit and replace with panic for now.. 
Once it is clarified, I can create another PR with updated code.. :)

Reply:
Sorry for not being clear.

Let me try to recap.

General rules on when to use:

- `log::error!()` -> run-time error that you want to draw attention on, the application may not be working as expected
- `panic!()` -> an invariant has broken, the application must stop or bad things will surely happen
- `assert!()` -> an assertion was found to be invalid, there is no other option to stop or bad things will surely happen
- `process:exit()` -> never :-)

In these cases, I will go for the option `log::error!()` because you have run-time errors that do not cause bad things to happen.